src/main/java/org/sunbird/bpreports/service/BPReportsServiceImpl.java

```diff
@@ -0,0 +1,376 @@
+
+    private static final Logger logger = LoggerFactory.getLogger(BPReportsServiceImpl.class);
+
+    @Autowired
+    AccessTokenValidator accessTokenValidator;
+
+    @Autowired
+    private UserUtilityService userUtilityService;
+
+    @Autowired
+    private CassandraOperationImpl cassandraOperation;
+
+    @Autowired
+    Producer kafkaProducer;
+
+    @Autowired
+    CbExtServerProperties serverProperties;
+
+    @Autowired
+    StorageService storageService;
+
+    @Override
+    public SBApiResponse generateBPReport(Map<String, Object> requestBody, String authToken) {
+        SBApiResponse response = ProjectUtil.createDefaultResponse(Constants.BP_REPORT_GENERATE_API);
+        try {
+            String userId = accessTokenValidator.fetchUserIdFromAccessToken(authToken);
+            // String userId = "95a357c6-99d9-43d2-8d1b-42bea6f8c132";
+            if (StringUtils.isBlank(userId)) {
+                updateErrorDetails(response, "Invalid user ID from auth token.", HttpStatus.BAD_REQUEST);
+                return response;
+            }
+
+            String courseId = (String) requestBody.get(Constants.COURSE_ID);
+            String batchId = (String) requestBody.get(Constants.BATCH_ID);
+            String orgId = (String) requestBody.get(Constants.ORG_ID);
+            SBApiResponse errResponse = validateGenerateReportRequestBody(requestBody);
+            if (!ObjectUtils.isEmpty(errResponse)) {
+                return response;
+            }
+
+            Map<String, Map<String, String>> userInfoMap = new HashMap<>();
+            userUtilityService.getUserDetailsFromDB(
+                    Arrays.asList(userId),
+                    Arrays.asList(Constants.ROOT_ORG_ID, Constants.USER_ID), userInfoMap);
+            String userOrgId = userInfoMap.get(userId).get(Constants.ROOT_ORG_ID);
+            if (StringUtils.isBlank(userOrgId) || !userOrgId.equals(orgId)) {
+                updateErrorDetails(response, "Requested User is not belongs to same organisation.", HttpStatus.BAD_REQUEST);
+                return response;
+            }
+
+            Map<String, Object> keyMap = new HashMap<>();
+            keyMap.put(Constants.ORG_ID, orgId);
+            keyMap.put(Constants.COURSE_ID, courseId);
+            keyMap.put(Constants.BATCH_ID, batchId);
+
+            List<Map<String, Object>> existingReportDetails = cassandraOperation.getRecordsByProperties(Constants.KEYSPACE_SUNBIRD,
+                    Constants.BP_ENROLMENT_REPORT_TABLE, keyMap, null);
+
+            if (!CollectionUtils.isEmpty(existingReportDetails)) {
+                String status = (String) existingReportDetails.get(0).get(Constants.STATUS);
+                if (Constants.STATUS_IN_PROGRESS_UPPERCASE.equalsIgnoreCase(status)) {
+                    response.getParams().setStatus(Constants.SUCCESS);
+                    response.getParams().setErrmsg("Report generation is in-progress.");
```

Comment:
with status as success -- do not set errmsg value... 
add one more attribute in the response body -- status -- which is in-progress, completed, failed -- etc...